### PR TITLE
doc: Reword deprecation message of MetadataExt::as_raw_stat

### LINKED ISSUE #62711: Which is accessor method of MetadataExt?
https://github.com/rust-lang/rust/blob/4b65a86ebace8600c8e269e8bfe3365cdc460e68/src/libstd/os/linux/fs.rs#L36-L41

### PR DESCRIPTION
**Pull request by tesuji**
Closes #62711
r? @Mark-Simulacrum

### CODE REVIEW

File: src/libstd/os/linux/fs.rs [hunk 1]
```diff
@@ -34,9 +34,10 @@ pub trait MetadataExt {
     /// }
     /// ```
     #[stable(feature = "metadata_ext", since = "1.1.0")]
-    #[rustc_deprecated(since = "1.8.0",
-                       reason = "deprecated in favor of the accessor \
-                                 methods of this trait")]
+    #[rustc_deprecated(
+        since = "1.8.0",
+        reason = "other methods of this trait is now prefered"
```
Comment (Mark-Simulacrum) [inline src/libstd/os/linux/fs.rs]:
```suggestion
        reason = "other methods of this trait are now prefered"
```

**Review by Mark-Simulacrum — COMMENTED** [review]
r=me with nit fixed

**Comment by lzutao** [comment]
The CI is passed.

**Pull request merged**